fdbserver/ptxn/test/TestTLogServer.actor.cpp

```diff
@@ -848,6 +834,116 @@ TEST_CASE("/fdbserver/ptxn/test/read_persisted_disk_on_tlog") {
+		tLogInitializations.back().tlogGroups = pContext->groupsPerTLog[i];
+		UID tlogId = ptxn::test::randomUID();
+		UID workerId = ptxn::test::randomUID();
+		StringRef fileVersionedLogDataPrefix = "log2-"_sr;
+		StringRef fileLogDataPrefix = "log-"_sr;
+		ptxn::InitializePtxnTLogRequest req = tLogInitializations.back();
+		const StringRef prefix = req.logVersion > TLogVersion::V2 ? fileVersionedLogDataPrefix : fileLogDataPrefix;
+
+		std::unordered_map<ptxn::TLogGroupID, std::pair<IKeyValueStore*, IDiskQueue*>> persistentDataAndQueues;
+		for (ptxn::TLogGroup& tlogGroup : pContext->groupsPerTLog[i]) {
+			std::string filename =
+			    filenameFromId(req.storeType, folder, prefix.toString() + "test", tlogGroup.logGroupId);
+			IKeyValueStore* data = openKVStore(req.storeType, filename, tlogGroup.logGroupId, 500e6);
+			state IDiskQueue* queue = new InMemoryDiskQueue(tlogGroup.logGroupId);
+			ds[tlogGroup.logGroupId] = data;
+			persistentDataAndQueues[tlogGroup.logGroupId] = std::make_pair(data, queue);
+		}
+
+		actors.push_back(ptxn::tLog(persistentDataAndQueues,
+		                            makeReference<AsyncVar<ServerDBInfo>>(),
+		                            LocalityData(),
+		                            initializeTLog,
+		                            tlogId,
+		                            workerId,
+		                            false,
+		                            Promise<Void>(),
+		                            Promise<Void>(),
+		                            folder,
+		                            makeReference<AsyncVar<bool>>(false),
+		                            makeReference<AsyncVar<UID>>(tlogId)));
+		initializeTLog.send(tLogInitializations.back());
+		std::cout << "Recruit tlog " << i << " : " << tlogId.shortString() << ", workerID: " << workerId.shortString()
+		          << "\n";
+	}
+
+	// replace fake TLogInterface with recruited interface
+	std::vector<Future<ptxn::TLogInterface_PassivelyPull>> interfaceFutures(pContext->numTLogs);
+	for (i = 0; i < pContext->numTLogs; i++) {
+		interfaceFutures[i] = tLogInitializations[i].reply.getFuture();
+	}
+	std::vector<ptxn::TLogInterface_PassivelyPull> interfaces = wait(getAll(interfaceFutures));
+	for (i = 0; i < pContext->numTLogs; i++) {
+		*(pContext->tLogInterfaces[i]) = interfaces[i];
+	}
+
+	for (auto& [tLogGroupID, tLogGroupLeader] : pContext->tLogGroupLeaders) {
+		tLogGroupLeader = pContext->tLogInterfaces[groupToLeaderId[tLogGroupID]];
+	}
+
+	state IKeyValueStore* d = ds[pContext->storageTeamIDTLogGroupIDMapper[storageTeamID]];
+
+	state std::pair<std::vector<Standalone<StringRef>>, std::vector<Version>> res =
+	    wait(commitInjectReturnVersions(pContext, storageTeamID, pContext->numCommits));
+	state std::vector<Standalone<StringRef>> expectedMessages = res.first;
+	wait(verifyPeek(pContext, storageTeamID, pContext->numCommits));
+
+	// wait 1s so that actors who update persistent data can do their job.
+	wait(delay(1.5));
```

Comment:
Nit: the initial setup part is duplicated in many tests. Can you extract it into a common function? This probably will save you a lot of time in the long run.

You can do it in a separate PR.

Reply:
will do in another PR